### PR TITLE
Refactor frontend

### PR DESCRIPTION
**Pull request by alex-ju**
This PR refactors frontend code to:
- remove the deprecated `govuk_frontend_toolkit` gem
- replace hardcoded code with existing components from `govuk_publishing_component` (tabs, panel and table)
- replace the old slimmer layout `wrapper` with the latest `core_layout`
- converts partials into isolated components using current convention (subscribe, metadata, calendar)
- refactors SCSS to follow convention

Recommend reviewing commit-by-commit.

<details>
<summary>Bank Holidays Preview</summary>

### Before
bank-holidays-before

### After
bank-holidays-after

</details>

<details>
<summary>Clocks Change Preview</summary>

### Before
clocks-change-before

### After
clocks-change-after

</details>

<p>&nbsp;</p>

[Trello card](https://trello.com/c/1Wl76i1F)
[Tech debt trello card](https://trello.com/c/Dpe7MWQp)

### CODE REVIEW

**Comment by sihugh** [comment]
As you've removed govuk_frontend_toolkit, I wonder if it's now possible to swap out `sass-rails` for `sassc-rails` and drop the sass dependency?  The Ruby sass gem goes end of life in March 2019 and sassc seems like the anointed replacement unless we go all webpacker which seems like a bigger change right now.

http://sass.logdown.com/posts/7081811

**Comment by alex-ju** [comment]
@sihugh thanks for reviewing this! sure will look into replacing `sass-rails` and will document the calendar component

**Comment by sihugh** [comment]
Don't worry too much about the sassc thing.  I'd rather get this merged and do it separately if it's going to take any time.

File: app/views/components/docs/metadata.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+name: Metadata
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply app/views/components/docs/metadata.yml]: Updated to add an example and document the component.

**Comment by alex-ju** [comment]
@sihugh switching to `sassc-rails` proves to be a bit trickier than I thought.

I get the following error after the switch:
```
ActionView::Template::Error:
Error: error in C function image-url: undefined method `[]' for nil:NilClass on line 5 of 
../../.rvm/gems/ruby-2.5.3/gems/govuk_publishing_components-13.5.4/app/assets/
stylesheets/govuk_publishing_components/components/govspeak/_advisory.scss,
in function `image-url`
```

Will create a card to investigate this further as it's not only related this individual app.

**Pull request merged**